### PR TITLE
Add safeguard against stuck connection.

### PR DESCRIPTION
**Pull request by rousveiga**
Fixes https://github.com/marvinroger/async-mqtt-client/issues/248, fix by https://github.com/mcspr/async-mqtt-client/commit/c1fcfd1.

### CODE REVIEW

**Comment by bertmelis** [comment]
Could you try the following:
Set the timeout to 10sec in the _onConnect ad reset in _clear and _onData (maybe only when CONNACK is received)

The way you did it now is on every outgoing message.

File: src/AsyncMqttClient.cpp [hunk 1]
```diff
@@ -378,6 +381,7 @@ void AsyncMqttClient::_addFront(AsyncMqttClientInternals::OutPacket* packet) {
     packet->next = _head;
   }
   _head = packet;
+  _client.setRxTimeout(10);
```
Comment (bertmelis) [inline src/AsyncMqttClient.cpp:384]: Better to move this line to 'connect'. We don't want to set the timeout on every outgoing packet.

**Review by bertmelis — CHANGES_REQUESTED** [review]

File: src/AsyncMqttClient.cpp (hunk 1 above)
Comment (bertmelis) [reply src/AsyncMqttClient.cpp:384]: Oh, and what about using the keepalive value as timeout (seconds!) instead of hardcode to 10 seconds?

**Comment by rousveiga** [comment]
Like this?

**Comment by bertmelis** [comment]
> Like this?

Yes, I assume it is still working?

**Review by bertmelis — APPROVED** [review]

**Review by vuthuy6586 — APPROVED** [review]

**Comment by rousveiga** [comment]
> Yes, I assume it is still working?

Sorry for the delay, yes it works fine!

**Pull request merged**